baggage_setter.go

```diff
@@ -22,58 +22,45 @@ package jaeger
 
 import (
 	"github.com/opentracing/opentracing-go/log"
+
+	"github.com/uber/jaeger-client-go/internal/baggage"
 )
 
 // baggageSetter is an actor that can set a baggage value on a Span given certain
 // restrictions (eg. maxValueLength).
-type baggageSetter interface {
-	setBaggage(span *Span, key, value string)
-}
-
-// defaultBaggageSetter sets the baggage key:value on the span while respecting the
-// maxValueLength and truncating the value if too long.
-type defaultBaggageSetter struct {
-	maxValueLength int
-	metrics        *Metrics
+type baggageSetter struct {
+	restrictionManager baggage.RestrictionManager
+	metrics            *Metrics
 }
 
-func newDefaultBaggageSetter(maxValueLength int, metrics *Metrics) *defaultBaggageSetter {
-	return &defaultBaggageSetter{
-		maxValueLength: maxValueLength,
-		metrics:        metrics,
+func newBaggageSetter(restrictionManager baggage.RestrictionManager, metrics *Metrics) baggageSetter {
+	return baggageSetter{
+		restrictionManager: restrictionManager,
+		metrics:            metrics,
 	}
 }
 
-func (s *defaultBaggageSetter) setBaggage(span *Span, key, value string) {
+func (s baggageSetter) setBaggage(span *Span, key, value string) {
```

Comment:
why do you pass the receiver by value?  

https://gist.github.com/yurishkuro/b135e6750df0f95aa86c35a77cc736d4